pkg/tools/prefixpool/prefixpool.go

```diff
@@ -505,12 +505,50 @@ func releasePrefixes(prefixes []string, released ...string) (remaining []string,
 					result = append(result, v.String())
 				}
 			}
+			result, _ = removeNestedNetworks(result)
 			return result, nil
 		}
 		prefixByPrefixLen = newPrefixByPrefixLen
 	}
 }
 
+func removeNestedNetworks(prefixes []string) ([]string, error) {
+	newPrefixes := make(map[string]struct{})
+	intersected := false
+	for newPrefixIndex, newPrefix := range prefixes {
+		intersected = false
+		_, newPrefixSubnet, err := net.ParseCIDR(newPrefix)
+		if err != nil {
+			logrus.Errorf("Wrong CIDR: %v", newPrefix)
+			return nil, err
+		}
+		for prefixIndex, prefix := range prefixes {
+			if prefixIndex == newPrefixIndex {
+				continue
+			}
+			_, prefixSubnet, _ := net.ParseCIDR(prefix)
```

Comment:
As I can see ParseCIDR is using here in n^2 times. We can parse prefixes once before comparing to reduce using from n^2 to n